spec/models/ems_event_spec.rb

```diff
@@ -157,6 +157,40 @@ def mock_raw_event_vm(raw_event)
     end
   end
 
+  context ".process_middleware_entities_in_event!" do
+    let(:middleware_ref) { "hawkular-test-path" }
+
+    before :each do
```

Comment:
I always try to use `let` for setting up objects that need to be tested against and `before :each` for setting about e.g. required side conditions.
So:
```ruby
let(:ems) { FactoryGirl.create(:ems_hawkular) }
let(:middleware_server) { FactoryGirl.create(:middleware_server, ems: ems) }
let(:event_hash) { {:middleware_type => 'MiddlewareServer', :ems_id => ems.id } }
```

This also helps future developers not to think about the before block when adding a test that does not need this setup